libhdf5/hdf5var.c

```diff
@@ -1000,6 +1000,15 @@ nc_def_var_szip(int ncid, int varid, int options_mask, int pixels_per_block)
     var->options_mask = options_mask;
     var->pixels_per_block = pixels_per_block;
 
+    /* We have to rememeber this in the filter info, or else
+     * nc_inq_var_szip() will not work properly. */
+    var->filterid = HDF5_FILTER_SZIP;
```

Comment:
@DennisHeimbigner note this hack. I have to put the szip info into the filter fields of the var info struct, in order to make nc_inq_var_szip() work. 

When you support multiple filters, this will have to be changed to set just one filter, instead of setting the only filter.